File: Core/value/integervalue.cpp
#include "integervalue.h"

IntegerValue::IntegerValue() : ValueBase() {
}

IntegerValue::IntegerValue(ValueGroup *valueGroup, QString id, VALUE_TYPE valueType, bool alwaysEmit, QObject *parent) : ValueBase(valueGroup, id, valueType, alwaysEmit, QVariant::Int, parent)
{

}

QVariant IntegerValue::_updateValue(QVariant newValue) {
    if (newValue.canConvert(QVariant::Int)) {
        return QVariant::fromValue(newValue);
    } else {
        return QVariant();
    }
}

File: AndroidApp/OshCore/src/main/java/com/osh/value/IntegerValue.java
package com.osh.value;

import com.osh.SerializationSupport;

public class IntegerValue extends ValueBase<IntegerValue, Integer> implements SerializationSupport {

	@Override
	protected Integer _updateValue(Object newValue) {
		if (newValue instanceof Integer) {
			return (Integer) newValue;
		} else if (newValue instanceof Number) {
			return ((Number) newValue).intValue();
		} else if (newValue instanceof String) {
			return Integer.parseInt((String) newValue);
		} else {
			return null;
		}
	}

	public IntegerValue(ValueGroup valueGroup, String id, ValueType valueType) {
		super(valueGroup, id, valueType);
	}

	public IntegerValue() {
	}

}
